File: ws/Perez/U2/WS23UnitTesting/ComplexOperations/src/test/java/ec/edu/espe/complexoperations/controller/OperationTest.java
package ec.edu.espe.complexoperations.controller;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
 *
 * @author Carlos Perez,The Javas, DCCO-ESPE
 */
public class OperationTest {
    
    public OperationTest() {
    }

    /**
     * Test of addTwoNumbers method, of class Operation.
     */
    @Test
    public void testAddTwoNumbers() {
        System.out.println("addTwoNumbers");
        float addend1 = 4.09F;
        float addend2 = 3.2F;
        float expResult = 7.29F;
        float result = Operation.addTwoNumbers(addend1, addend2);
        assertEquals(expResult, result, 0);

    }
@Test
public void testAddOneNegativeNumberAndOnePositive(){
        System.out.println("addTwoNumbers");
        float addend1 = 1.2F;
        float addend2 = -2.3F;
        float expResult = -1.1F;
        float result = Operation.addTwoNumbers(addend1, addend2);
        assertEquals(expResult, result, 0);
}
    
}

File: hw/Pérez/U2/HW17TestCaseSolution/Test/Operation.test.js
import { Operation } from '../ec.edu.espe.complexoperations.controller/Operation.js';


test('addTwoNumbers adds positive numbers', () => {
    const addend1 = 4.09;
    const addend2 = 3.2;
    const result = Operation.addTwoNumbers(addend1, addend2);
    expect(result).toBeCloseTo(7.29);
});

test('addTwoNumbers adds one negative number and one positive', () => {
    const addend1 = 1.2;
    const addend2 = -2.3;
    const result = Operation.addTwoNumbers(addend1, addend2);
    expect(result).toBeCloseTo(-1.1);
});